### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by jsanda**
Update overview and add some developer notes

┆Issue is synchronized with this [Jiraserver Task](https://k8ssandra.atlassian.net/browse/K8SSAND-755) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-755
┆Priority: Medium

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+* kind >= 0.11.1
+* Docker
+
+## Type definitions
+The Go type definition for custom resources live under the `api` directory in files with a `_types.go` suffix. The CRDs are derived from the structs defined in these files.
+
+## Updating CRDs
+As mentioned previously, the CRDs are generated based off the contents of the `_types.go` files. The CRDs live under `config/crd`.
+
+Run `make manifest` update CRDs.
```
Comment (adutra) [inline README.md:39]:
```suggestion
Run `make manifest` to update CRDs.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,74 @@
...
+
+**Note:** Any changes to the `_types.go` files should be followed by `make manifests`
```
Comment (adutra) [inline README.md:41]:
```suggestion
**Note:** Any changes to the `_types.go` files should be followed by `make manifests`.
```

Comment (adutra) [reply README.md:41]: Shouldn't we also do `make generate` when we change the API types?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,74 @@
...
+
+To set up two kind clusters for multi-cluster tests run the following:
+
+`scripts/setup-kind-multicluster.sh --clusters 2`
+
+### Resource Requirements
+Multi-cluster tests will be more resource intensive than other tests. The Docker VM on my MacBook Pro is configured with 6 CPUs and  10 GB of memory for these tests. Your mileage may vary on other operating systems/setups. 
+
+
+If you are not, please start by checking out the docs at [k8ssandra.io](https://k8ssandra.io/). At
```
Comment (adutra) [inline README.md:74]: This looks like a dangling line.

**Review by adutra — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jsanda) [reply README.md:41]: good catch

**Pull request merged**